src/components/NftPreview/NftPreview.tsx

```diff
@@ -122,15 +109,41 @@ function NftPreview({ galleryNftRef }: Props) {
     [nft.contractAddress]
   );
 
+  const {
+    pathname,
+    query: { username, collectionId },
+  } = useRouter();
+
+  // whether the user is on a gallery page or collection page prior to clicking on an NFT
+  const originPage = collectionId ? 'collection' : 'gallery';
+
   return (
-    <StyledNftPreview
-      maxWidth={nftPreviewMaxWidth}
-      width={nftPreviewWidth}
-      backgroundColorOverride={backgroundColorOverride}
+    <Link
+      // path that will be shown in the browser URL bar
+      as={`/${username}/${collection.dbid}/${nft.dbid}`}
+      // query params purely for internal tracking. this will NOT be displayed in URL bar.
+      // the path will either be `/[username]` or `/[username]/[collectionId]`, with the
+      // appropriate query params attached. this allows the app to stay on the current page,
+      // while also feeding the modal the necessary data to display an NFT in detail.
+      href={`${pathname}?username=${username}&collectionId=${collection.dbid}&nftId=${nft.dbid}&originPage=${originPage}&modal=true`}
+      // from the docs: if the child of Link is a custom component that wraps an <a> tag,
+      // you must add `passHref` to Link. This is necessary if you’re using libraries like
+      // styled-components. Without this, the <a> tag will not have the href attribute,
+      // which hurts your site's accessibility and might affect SEO
+      passHref
+      // disable scroll-to-top when the modal opens
+      scroll={false}
```

Comment:
this is the key change that allows:
1. the app to think that it's on the same page when opening the modal
2. the URL to _appear_ updated to the user